content/en/events/gsoc-2024.md

```diff
@@ -0,0 +1,225 @@
+* See Kubeflow Notebooks 2.0 design document: [https://docs.google.com/document/d/1_zk06zebbaTBdJ8TdU07Ibky25hqHGARXjVcsp2qEnU/edit](https://docs.google.com/document/d/1_zk06zebbaTBdJ8TdU07Ibky25hqHGARXjVcsp2qEnU/edit)
+
+**Skills required:** Kubernetes Controllers (Golang - Kubebuilder) AND/OR Web Development (JS - Angular, Python - Flask)
+
+**Difficulty:** medium/high
+
+**Length:** 250 hrs
+
+**Mentors:** Mathew Wicks, Kimonas Sitorchos, Julius von Kohout
+
+---
+---
+
+###  Project 2: Rootless Kubeflow Container Images (Istio Ambient Mesh)
+
+Kubeflow uses Istio as a service mesh, which by default requires “root level” network permissions for its init-containers. We want to reduce the number of privileged containers required to run Kubeflow, so are investigating using the Istio CNI, and eventually the Istio Ambient mesh.
+
+You would be involved in testing and investigating the impacts of these changes, and helping push the integration forwards.
+
+See the proposal for more information: [https://github.com/kubeflow/manifests/blob/master/proposals/20200913-rootlessKubeflow.md](https://github.com/kubeflow/manifests/blob/master/proposals/20200913-rootlessKubeflow.md)
+
+**Skills required:** Istio, Kubernetes, YAML
+
+**Difficulty:** medium
+
+**Length:** 250 hrs
+
+**Mentors:** Kimonas Sitorchos, Julius von Kohout
+
+---
+---
+
+###   Project 3: Triage and Categorize Kubeflow GitHub Issues & PRs
+
+The Kubeflow project needs help to triage, categorize, and highlight important Issues/PRs from the [https://github.com/kubeflow/kubeflow](https://github.com/kubeflow/kubeflow) GitHub repo.  There are around 200 open Issues and 200 open PRs, in addition to many Issues/PRs that have been lost to time (closed automatically due to inactivity).
+
+Specifically, your goal would be to:
+
+* Decide which Issues/PRs are still relevant
+* Categorize Issues/PRs by type
+* De-duplicate multiple Issues for the same request
+* Suggest which ones are the most important.
+* Help find “good first issues” for new members:
+    * [https://github.com/kubeflow/manifests/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22](https://github.com/kubeflow/manifests/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+* Review which PRs are likely safe to merge (especially dependabot ones)
+
+**Skills required:** GitHub, Kubernetes, YAML, Python, GO, JS
+
+**Difficulty:** medium
+
+**Length:** 250 hrs
+
+**Mentors:** Mathew Wicks, Kimonas Sitorchos, Julius von Kohout
+
+---
+---
+
+###  Project 4: Implement LLM Tuning API for Katib
+
+Recently, we implemented [https://github.com/kubeflow/training-operator/blob/master/docs/proposals/train_api_proposal.md](a new `train` Python SDK API) in Kubeflow Training Operator to easily fine-tune LLMs on multiple GPUs with predefined datasets provider, model provider, and HuggingFace trainer.
```

Comment:
@rareddy This link is around the wrong way, it should be `[a new `train` Python SDK API](https://github.com/kubeflow/training-operator/blob/master/docs/proposals/train_api_proposal.md)`